src/collectives/device/all_reduce.h

```diff
@@ -115,7 +117,29 @@ __device__ void ncclAllReduceKernel(struct CollectiveArgs* args) {
     slice = ring->devUserRanks[0];
     offset = chunkOffset + slice * chunkSize;
     maxOffset = min(chunkSize, size-offset);
+    
+#if 1
+    //my reduce
+    Prims::Reduce(tid, nthreads,
+          prevInput  + poffset,
+          thisInput  + offset,
+    //	  nextdirect ? (sharedNextOutput + offset) : (nextOutput + noffset),
+		  prevInput + poffset,
+          sliceSize, maxOffset,
+          step,
+          waitDoneFromNext, waitReadyFromPrev,
+          postReadyToNext, postDoneToPrev);
 
+    Prims::Copy(tid, nthreads,
```

Comment:
@Sergei-Lebedev 
What's the rationale here?
note there was no call to
NEXT_STEPS,
so the flags used here are the same flags as used by the previous call on line 121

Reply:
Need to remove this call, it's unnecessary and left from my previous tests